### ISSUE TITLE
Read in data from LJH file for inverted detectors

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#127]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/127) by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Some people @danbek apparently wire up their detectors backwards and want MASS to be able to deal with it.

I think this should be straightforward to do.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

@oneilg was interested in this issue, too.

**Comment by joefowler** [comment]
**Original comment by Joseph Adams (Bitbucket: [Joe Adams](https://bitbucket.org/Joe Adams), ).**

----------------------------------------

Yep Steve Smith hacked this in as well, would be a nice feature.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Joe: thanks so much for commenting on this! Galen and I were having no luck in recalling when and from whom we had heard this very complaint before. You are right, it was @smithsj2 who needed this feature.

I think I have it ready to go, but I left the office before I finished committing. The update should be ready tomorrow.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Added new feature to allow data invert-on-read.

If you have a channel that was wired up backwards, so that pulses are
negative-going, then you can fix this in analysis by saying:


```
#!python

for channum in [5,17,23,25]:
    ds = data.channel[channum]
    ds.invert_data = True
	ds.clear_cache() # prevent already-read data from being used

```

This will mean that all future reads of raw data will get the unit16-
inverse of the data actually recorded to the LJH file. Fixes #127.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Removing milestone: v0.6 (automated comment)
